src/protocols/hii_string.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2019 Intel Corporation
+// Copyright (c) Microsoft Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(unused)]
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0xfd96974,
+    0x23aa,
+    0x4cdc,
+    0xb9,
+    0xcb,
+    &[0x98, 0xd1, 0x77, 0x50, 0x32, 0x2a],
+);
+
+#[repr(C)]
+pub struct Protocol {
+    pub new_string: eficall! {fn(
+        *mut Protocol,
+        crate::efi::protocols::hii_database::HiiHandle,
```

Comment:
I usually drop the `::efi::` part, since that is really just for compatibility to the tianocore interfaces. So this could be `crate::protocols::hii_database::HiiHandle`. Same for all instances below.